classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,482 @@
+
+# ### (Neptune) Track datasets using Neptune
+# Since this dataset will be used among all the runs in the project, we track it at the project level
+
+# In[4]:
+
+
+project["data/keras/files"].track_files(
+    "s3://neptune-examples/data/text-classification/aclImdb_v1.tar.gz"
+)
+project.sync()
+
+
+# ### (Neptune) Download files from S3 using Neptune
+
+# In[5]:
+
+
+print("Downloading data...")
+project["data/keras/files"].download("..")
+
+
+# ### Extract downloaded files
+
+# In[6]:
+
+
+import tarfile
+
+print("Extracting data...")
+my_tar = tarfile.open("../aclImdb_v1.tar.gz")
+my_tar.extractall("..")
+my_tar.close()
+
+
+# ### Remove `unsup` subfolder and rename `aclImdb` folder
+
+# In[7]:
+
+
+import shutil
+
+shutil.rmtree("../aclImdb/train/unsup")
+os.remove("../aclImdb_v1.tar.gz")
+
+if os.path.exists("../data"):
+    shutil.rmtree("../data")
+os.rename("../aclImdb", "../data")
+
+
+# (Neptune) Upload dataset sample to Neptune project
+
+# In[8]:
+
+
+import random
+
+project["data/keras/sample/train/pos"].upload(
+    f"../data/train/pos/{random.choice(os.listdir('../data/train/pos'))}"
+)
+project["data/keras/sample/train/neg"].upload(
+    f"../data/train/neg/{random.choice(os.listdir('../data/train/neg'))}"
+)
+project["data/keras/sample/test/pos"].upload(
+    f"../data/test/pos/{random.choice(os.listdir('../data/test/pos'))}"
+)
+project["data/keras/sample/test/neg"].upload(
+    f"../data/test/neg/{random.choice(os.listdir('../data/test/neg'))}"
+)
+
+
+# ### Generate training, validation, and test datasets
+
+# In[44]:
+
+
+data_params = {
+    "batch_size": 32,
+    "validation_split": 0.2,
+    "max_features": 20000,
+    "embedding_dim": 128,
+    "sequence_length": 500,
+    "seed": 42,
+}
+
+
+# (Neptune) Log data metadata to Neptune
+
+# In[45]:
+
+
+run = neptune.init_run(name="Keras text classification", tags=["keras"])
+
+
+# In[46]:
+
+
+run["data/params"] = data_params
+
+
+# In[47]:
+
+
+import re
+import string
+
+from tensorflow.keras.layers import TextVectorization
+
+# In[48]:
+
+
+raw_train_ds, raw_val_ds = tf.keras.preprocessing.text_dataset_from_directory(
+    "../data/train",
+    batch_size=data_params["batch_size"],
+    validation_split=data_params["validation_split"],
+    subset="both",
+    seed=data_params["seed"],
+)
+
+raw_test_ds = tf.keras.preprocessing.text_dataset_from_directory(
+    "../data/test", batch_size=data_params["batch_size"]
+)
+
+print(f"Number of batches in raw_train_ds: {raw_train_ds.cardinality()}")
+print(f"Number of batches in raw_val_ds: {raw_val_ds.cardinality()}")
+print(f"Number of batches in raw_test_ds: {raw_test_ds.cardinality()}")
+
+
+# ### Previewing data
+
+# In[49]:
+
+
+for text_batch, label_batch in raw_train_ds.take(1):
+    for i in range(5):
+        print(text_batch.numpy()[i])
```

Comment:
Probably add some context in the print like `print("Sample {i}:", text_batch.numpy()[i])`

But do we really require this print for the script?